app/notification.go

```diff
@@ -164,6 +164,16 @@ func SendNotifications(post *model.Post, team *model.Team, channel *model.Channe
 				}
 			}
 
+			//If email verification is required and user email is not verified don't send email.
+			emailVerified := true
+			l4g.Debug("Server requires email verification: %v", utils.Cfg.EmailSettings.RequireEmailVerification)
+			l4g.Debug("User email verification status: %v", profileMap[id].EmailVerified)
+
+			if utils.Cfg.EmailSettings.RequireEmailVerification && !profileMap[id].EmailVerified {
+				l4g.Error("Skipped sending notification email to %v, address not verified. [details: user_id=%v]", profileMap[id].Email, id)
+				emailVerified = false
```

Comment:
I think you can just use a continue statement here to save us having to get the user's status below

Reply:
Complete